src/StorageFolding.cpp

```diff
@@ -150,17 +198,12 @@ class AttemptStorageFoldingOfFunction : public IRMutator {
         }
 
         // Any folds that took place folded dimensions away entirely, so we can proceed recursively.
-        if (const For *f = result.as<For>()) {
-            Stmt body = mutate(f->body);
-            if (body.same_as(f->body)) {
-                stmt = result;
-            } else {
-                stmt = For::make(op->name, op->min, op->extent, op->for_type, op->device_api, body);
-            }
+        body = mutate(body);
+        if (body.same_as(op->body)) {
```

Comment:
I think this is OK... I'm basically checking to see if we should return the original for loop unchanged or not. The mutate here might change it, or the one on line 178, or the ones that add assert statements to the loop body. If any of them change it, we need to make a new for loop.